bgpd/bgp_clist.c

```diff
@@ -718,125 +718,137 @@ community_regexp_delete (struct community *com, regex_t * reg)
 
 /* When given community attribute matches to the community-list return
    1 else return 0.  */
-int community_list_match(struct community *com, struct community_list *list)
+bool community_list_match(struct community *com, struct community_list *list)
 {
 	struct community_entry *entry;
 
 	for (entry = list->head; entry; entry = entry->next) {
 		if (entry->any)
-			return entry->direct == COMMUNITY_PERMIT ? 1 : 0;
+			return entry->direct == COMMUNITY_PERMIT ? true : false;
```

Comment:
As I previously suggested it could be just
`return entry->direct == COMMUNITY_PERMIT;`